CodingPractice/PythonAssignments/cleancode/EmailExtract.py

```diff
@@ -0,0 +1,30 @@
+import re
+
+
+class MalformedEmailAddressException(Exception):
+    pass
+
+
+class EmailDetails:
+    def __init__(self, _id, _domain):
+        self.id = _id
+        self.domain = _domain
+
+    def __eq__(self, other):
+        return self.id == other.id and self.domain == other.domain
+
+    def get_id(self):
+        return self.id
+
+    def get_domain(self):
+        return self.domain
+
+
+def extract_email_address_details(email_addr):
+    try:
+        if not re.match(r'[^@]+@[^@]+\.[^@]+', email_addr):
```

Comment:
Oh man yes that is one way to do it but using a regex is no a good idea here as they are very hard to understand without spending time working out what they do. Regex isonly really used for matching across large blocks of text or as a last resort. Did you actually read the section I mentioned in the comment below ?!?